lib/hanami/application/slice_registrar.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+require_relative "../constants"
+require_relative "../slice"
+
+module Hanami
+  class Application
+    # @api private
+    class SliceRegistrar
+      attr_reader :application, :slices
+      private :application, :slices
+
+      def initialize(application)
+        @application = application
+        @slices = {}
+      end
+
+      def register(name, slice_class = nil, &block)
+        if slices.key?(name.to_sym)
+          raise SliceLoadError, "Slice '#{name}' is already registered"
+        end
+
+        # TODO: raise error unless name meets format (i.e. single level depth only)
+
+        slices[name.to_sym] = slice_class || build_slice(name, &block)
+      end
+
+      def [](name)
+        slices.fetch(name) do
+          raise SliceLoadError, "Slice '#{name}' not found"
+        end
+      end
+
+      def freeze
+        slices.freeze
+        super
+      end
+
+      def load_slices
+        slice_configs = Dir[root.join(CONFIG_DIR, "slices", "*#{RB_EXT}")]
+          .map { |file| File.basename(file, RB_EXT) }
+
+        slice_dirs = Dir[File.join(root, SLICES_DIR, "*")]
+          .select { |path| File.directory?(path) }
+          .map { |path| File.basename(path) }
+
+        (slice_dirs + slice_configs).uniq.sort.each do |slice_name|
+          load_slice(slice_name)
+        end
+
+        self
+      end
+
+      def each(&block)
+        slices.each_value(&block)
+      end
```

Comment:
I'm upgrading to alpha7 for my project and now cannot run `hanami console`.
I found this method behaves differently, therefore, breaks hanami-cli. https://github.com/hanami/cli/blob/v2.0.0.alpha6.1/lib/hanami/console/plugins/slice_readers.rb#L15
I'm not sure how to best fix this. Can you help resolve this?